pkg/printers/interface.go

```diff
@@ -94,3 +94,12 @@ type ErrNoDescriber struct {
 func (e ErrNoDescriber) Error() string {
 	return fmt.Sprintf("no describer has been defined for %v", e.Types)
 }
+
+// OutputOptions represents resource output options which is used to generate a resource printer.
+type OutputOptions struct {
+	FmtType string // e.g. json, yaml, template, jsonpath etc.
```

Comment:
I agree. Unfortunately, these are not defined as constant types in the code. For now, I have mentioned the filename where the supported format types are listed inside a function. I would like to make a separate change for defining these valid format types as constants.